pkg/services/serviceaccounts/api/token.go

```diff
@@ -110,6 +111,42 @@ func (api *ServiceAccountsAPI) ListTokens(ctx *contextmodel.ReqContext) response
 	return response.JSON(http.StatusOK, result)
 }
 
+// @PERCONA
+// swagger:route GET /auth/serviceaccount serviceaccounts retrieveServiceAccount
+//
+// # CurrentServiceAccount get current service account info
+//
+// Requires service account token authentication.
+//
+// Responses:
+// 200: retrieveServiceAccountResponse
+// 400: badRequestError
+// 401: unauthorisedError
+// 403: forbiddenError
+// 500: internalServerError
+func (api *ServiceAccountsAPI) CurrentServiceAccount(ctx *contextmodel.ReqContext) response.Response {
+	if !ctx.IsServiceAccount {
+		serviceAccountID, err := api.service.MigrateApiKey(ctx.Req.Context(), ctx.OrgID, ctx.ApiKeyID)
```

Comment:
are you sure that we still need to migrate API keys manually? isn't it supposed to be migrated by Grafana?

Reply:
in case it's migrated by grafana on the first start then we can remove a lot of our code from this PR

Reply:
You are right and Grafana 10 should care of this. Once I will verify it I will remove all code related to manual migration on our side.